File: src/main/java/com/donniemattingly/kotlinbrownbag/RecipeRepository.java
package com.donniemattingly.kotlinbrownbag;

import org.springframework.stereotype.Repository;

import java.util.Collections;
import java.util.List;

@Repository
public class RecipeRepository {

  List<Recipe> getRecipesByType(String type) {
    return Collections.emptyList();
  }

  List<Recipe> getAllRecipes(){
    return Collections.emptyList();
  }

}

File: src/main/java/com/donniemattingly/kotlinbrownbag/RecipeRepository.kt
package com.donniemattingly.kotlinbrownbag

import org.springframework.stereotype.Repository

import java.util.Collections

@Repository
class RecipeRepository {

    val allRecipes: List<Recipe>
        get() = emptyList()

    fun getRecipesByType(type: String): List<Recipe> {
        return emptyList()
    }

}
